### PR TITLE
Fix double indexing which created unnecessary vectors, remove curly braces syntax and use generators

### PR DESCRIPTION
**Pull request by chriscoey**

### CODE REVIEW

**Comment by chriscoey** [comment]
I need to go update the require and yml files

File: src/conic_algorithm.jl [hunk 1]
```diff
@@ -105,6 +106,8 @@ type PajaritoConicModel <: MathProgBase.AbstractConicModel
     # MIP constructed data
     model_mip::JuMP.Model       # JuMP MIP (outer approximation) model
     x_mip::Vector{JuMP.Variable} # JuMP vector of original variables
+    x_int::Vector{JuMP.Variable} # JuMP vector of integer variables
```
Comment (mlubin) [inline src/conic_algorithm.jl]: Maybe call this a subvector to make clear that it's not distinct from ``x_mip``?

Comment (chriscoey) [inline src/conic_algorithm.jl]: changed x_mip to x_all, noted others are subvectors

File: src/conic_algorithm.jl [hunk 2]
```diff
@@ -86,6 +86,7 @@ type PajaritoConicModel <: MathProgBase.AbstractConicModel
     c_new::Vector{Float64}      # Objective coefficient subvector for continuous variables
     A_new::SparseMatrixCSC{Float64,Int64} # Submatrix of A containing full rows and continuous variable columns
     keep_cols::Vector{Int}      # Indices of variables in original problem that remain after removing zeros
+    keep_var_types::Vector{Symbol} # Variable types vector on variables after transforming
```
Comment (chriscoey) [inline src/conic_algorithm.jl]: i use it in both mip constructor and conic constructor functions, but don't need to keep it around after that. but there are a bunch of other data variables i save but don't need after constructing (eg row_to_slckj and row_to_slckv, and all the original data) too, i just need to share them between the constructor functions

Comment (mlubin) [inline src/conic_algorithm.jl]: I'd argue that it's better to recompute them in order to keep the model object simpler.

**Comment by chriscoey** [comment]
Compromise: I can return them and manage them locally in the optimize
function

**Pull request merged**